src/app/common/services/ai-skills.service.ts

```diff
@@ -17,42 +17,41 @@ export class AiSkillsService extends BaseHttpApiService {
 		super(loginService, http, configService, messageService);
 	}
 
-    // Inspired by
-    // - https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem
-    // - https://stackoverflow.com/questions/5234581/base64url-decoding-via-javascript
-    toBase64Url(text: string): string {
-        // To allow for unicode characters, convert it to a binary string first
-        const bytes = new TextEncoder().encode(text);
-        const binString = Array.from(bytes, (byte) =>
-                                                 String.fromCodePoint(byte),
-                                                ).join("");
-        const base64 = btoa(binString);
-        // To make it url safe, replace + and /
-        return base64.replace('/\+/g', '-').replace('/\//g', '_');
-    }
+	// Inspired by
+	// - https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem
+	// - https://stackoverflow.com/questions/5234581/base64url-decoding-via-javascript
+	toBase64Url(text: string): string {
+		// To allow for unicode characters, convert it to a binary string first
+		const bytes = new TextEncoder().encode(text);
+		const binString = Array.from(bytes, (byte) => String.fromCodePoint(byte)).join('');
+		const base64 = btoa(binString);
+		// To make it url safe, replace + and /
+		return base64.replace('/+/g', '-').replace('///g', '_');
+	}
 
 	getAiSkillsResult(textToAnalyze: string): Promise<AiSkillsResult> {
+		const encodedText = encodeURIComponent(textToAnalyze);
```

Comment:
I don't think this can handle unicode characters properly. Also it shouldn't be necessary if the `toBase64Url` works properly